src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -36,12 +39,16 @@ internal partial class NavigationBarController : ForegroundThreadAffinitizedObje
         private readonly IAsynchronousOperationListener _asyncListener;
 
         private bool _disconnected = false;
-        private Workspace? _workspace;
 
         /// <summary>
-        /// Latest model and selected items produced once <see cref="DetermineSelectedItemInfoAsync"/> completes and
-        /// presents the single item to the view.  These can then be read in when the dropdown is expanded and we want
-        /// to show all items.
+        /// The last fully computed model.
+        /// </summary>
+        private NavigationBarModel _model_OnlyAccessOnUIThread;
```

Comment:
This gets cleaner once https://github.com/dotnet/roslyn/pull/54443 goes in.  Then we don't need to have state for this and can just pass this value through our work queue.